lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/TransactionPayload.kt

```diff
@@ -199,11 +186,34 @@ data class DeclareTransactionV3Payload(
     val feeDataAvailabilityMode: DAMode,
 
     @SerialName("version")
-    val version: Felt = Felt(3),
+    val version: Felt,
 
     @SerialName("type")
     override val type: TransactionType = TransactionType.DECLARE,
-) : DeclareTransactionPayload()
+) : DeclareTransactionPayload() {
+    constructor(
+        contractDefinition: Cairo1ContractDefinition,
+        nonce: Felt,
+        signature: Signature,
+        senderAddress: Felt,
+        compiledClassHash: Felt,
+        resourceBounds: ResourceBoundsMapping,
+        version: Felt,
+    ) : this(
+        contractDefinition = contractDefinition,
+        nonce = nonce,
+        signature = signature,
+        senderAddress = senderAddress,
+        compiledClassHash = compiledClassHash,
+        resourceBounds = resourceBounds,
+        tip = Uint64.ZERO,
+        paymasterData = emptyList(),
+        accountDeploymentData = emptyList(),
+        nonceDataAvailabilityMode = DAMode.L1,
+        feeDataAvailabilityMode = DAMode.L1,
+        version = version,
```

Comment:
Should I change this to:
```suggestion
        tip = TransactionFactory.tip,
        paymasterData = TransactionFactory.paymasterData,
        accountDeploymentData = TransactionFactory.accountDeploymentData,
        nonceDataAvailabilityMode = TransactionFactory.nonceDataAvailabilityMode,
        feeDataAvailabilityMode = TransactionFactory.feeDataAvailabilityMode,
```
I don't like the idea of relying on another object, but I don't like defaults scattered around also

Reply:
I'd leave this as is right now tbh.